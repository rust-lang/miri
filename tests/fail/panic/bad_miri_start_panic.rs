//@compile-flags: -Zmiri-disable-abi-check
// This feature is required to trigger the error using the "C" ABI.
#![feature(c_unwind)]

extern "C" {
    fn miri_start_panic(payload: *mut u8) -> !;
}

fn main() {
    unsafe { miri_start_panic(&mut 0) }
    //~^ ERROR: unwinding past a stack frame that does not allow unwinding
}
