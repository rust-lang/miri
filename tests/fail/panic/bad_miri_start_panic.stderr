WARNING: the flag `-Zmiri-disable-abi-check` is deprecated and planned to be removed.
If you have a use-case for it, please file an issue.
error: Undefined Behavior: unwinding past a stack frame that does not allow unwinding
  --> $DIR/panic/bad_miri_start_panic.rs:LL:CC
   |
LL |     unsafe { miri_start_panic(&mut 0) }
   |              ^^^^^^^^^^^^^^^^^^^^^^^^ unwinding past a stack frame that does not allow unwinding
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `main` at $DIR/panic/bad_miri_start_panic.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

